server/game/cards/characters/02/croneofvaesdothrak.js

```diff
@@ -5,16 +5,16 @@ class CroneOfVaesDothrak extends DrawCard {
     setupCardAbilities(ability) {
         this.reaction({
             when: {
-                onCardDiscarded: (event, player, card, allowSave, originalLocation) =>
-                    ((originalLocation === 'hand' || originalLocation === 'draw deck')
-                     && card.getType() === 'character'
-                     && player !== this.controller)
+                onCardDiscarded: (event, params) =>
```

Comment:
How about something more like this? https://github.com/ystros/throneteki/compare/fix-cersei-finally-i-guess-sigh...ystros:merged-events?expand=1

**Pros:**
* Gets rid of the secondary `params` object by just merging it directly into the event.
* No longer need to access event.params directly in the handlers, so that can clean quite a bit up.
* Still fulfills all the requirements introduced by Tywin (replaceable event params).

**Cons:**
* We have two different ways of raising events now. I could make it do the merge if the first parameter has a special flag on it (e.g. `raiseEvent('onCardDiscarded', { merge: true, card: card })`), or just by detecting that it's a single non-class object parameter, but that feels a bit too magical. Thought it would be better to be explicit (in the short term while there's two ways, at least).
* Converting all events over (which should be done rather than maintain both forever) will be super painful. I'm not even sure there's a good migration path - just a catastrophic 'convert it all' commit for each event.

(those cons still existed with the `params` argument, it was just less explicit)